src/modules/wind_estimator/wind_estimator_main.cpp

```diff
@@ -235,23 +317,210 @@ WindEstimatorModule::Run()
 	}
 }
 
-void WindEstimatorModule::update_params()
+
+void AirspeedModule::update_params()
 {
 	updateParams();
 
-	// update wind & airspeed scale estimator parameters
-	_wind_estimator.set_wind_p_noise(_param_west_w_p_noise.get());
-	_wind_estimator.set_tas_scale_p_noise(_param_west_sc_p_noise.get());
-	_wind_estimator.set_tas_noise(_param_west_tas_noise.get());
-	_wind_estimator.set_beta_noise(_param_west_beta_noise.get());
-	_wind_estimator.set_tas_gate(_param_west_tas_gate.get());
-	_wind_estimator.set_beta_gate(_param_west_beta_gate.get());
+	/* update airspeedValidator parameters */
+	for (int i = 0; i < _number_of_airspeed_sensors; i++) {
+		_airspeed_validator[i].set_wind_estimator_wind_p_noise(_param_west_w_p_noise.get());
+		_airspeed_validator[i].set_wind_estimator_tas_scale_p_noise(_param_west_sc_p_noise.get());
+		_airspeed_validator[i].set_wind_estimator_tas_noise(_param_west_tas_noise.get());
+		_airspeed_validator[i].set_wind_estimator_beta_noise(_param_west_beta_noise.get());
+		_airspeed_validator[i].set_wind_estimator_tas_gate(_param_west_tas_gate.get());
+		_airspeed_validator[i].set_wind_estimator_beta_gate(_param_west_beta_gate.get());
+		_airspeed_validator[i].set_wind_estimator_scale_estimation_on(_param_west_scale_estimation_on.get());
+
+		/* Only apply manual entered airspeed scale to first airspeed measurement */
+		_airspeed_validator[0].set_airspeed_scale(_param_west_airspeed_scale.get());
+
+
+
+		_airspeed_validator[i].set_tas_innov_threshold(_tas_innov_threshold.get());
+		_airspeed_validator[i].set_tas_innov_integ_threshold(_tas_innov_integ_threshold.get());
+		_airspeed_validator[i].set_checks_fail_delay(_checks_fail_delay.get());
+		_airspeed_validator[i].set_checks_clear_delay(_checks_clear_delay.get());
+		_airspeed_validator[i].set_airspeed_stall(_airspeed_stall.get());
+	}
 }
 
-int WindEstimatorModule::custom_command(int argc, char *argv[])
+void AirspeedModule::poll_topics()
+{
+	const hrt_abstime time_now_usec = hrt_absolute_time();
+
+	if (_estimator_status_sub.update(&_estimator_status)) {
+		// _vehicle_attitude_valid = (_time_now_usec - sensor_bias.timestamp < 1_s) && (sensor_bias.timestamp > 0);
+	}
+
+	if (_sensor_bias_sub.update(&_sensor_bias)) {
+		// _vehicle_attitude_valid = (_time_now_usec - sensor_bias.timestamp < 1_s) && (sensor_bias.timestamp > 0);
+	}
+
+	if (_vehicle_air_data_sub.update(&_vehicle_air_data)) {
+		// vehicle_air_data_valid = (_time_now_usec - vehicle_air_data.timestamp < 1_s) && (vehicle_air_data.timestamp > 0);
+	}
+
+	if (_vehicle_attitude_sub.update(&_vehicle_attitude)) {
+		// _vehicle_attitude_valid = (_time_now_usec - vehicle_attitude.timestamp < 1_s) && (vehicle_attitude.timestamp > 0);
+	}
+
+	if (_vehicle_land_detected_sub.update(&_vehicle_land_detected)) {
+		// _vehicle_attitude_valid = (_time_now_usec - vehicle_attitude.timestamp < 1_s) && (vehicle_attitude.timestamp > 0);
+	}
+
+	if (_vehicle_local_position_sub.update(&_vehicle_local_position)) {
```

Comment:
@sfuhrer I think you should do all these updates and then at the very end check if the data is not older than 1 second. I'm not sure what the likelihood is but I guess it could be that the timestamp from the topic is newer than the one you are using to compare.